src/EditorFeatures/Test/CodeActions/CodeChangeProviderMetadataTests.cs

```diff
@@ -0,0 +1,188 @@
+            {
+                if (string.IsNullOrEmpty(namedGroup.Key))
+                {
+                    continue;
+                }
+
+                if (namedGroup.Take(2).Count() == 1)
+                {
+                    continue;
+                }
+
+                var providerNames = string.Join(", ", namedGroup.Select(exportedProvider => exportedProvider.Part.Definition.Type.FullName));
+                failureMessage.AppendLine($"'{namedGroup.Key}' is used by the following providers: {providerNames}");
+            }
+
+            Assert.True(failureMessage.Length == passLength, failureMessage.ToString());
+        }
+
+        [Theory]
+        [InlineData(typeof(CodeFixProvider), typeof(PredefinedCodeFixProviderNames))]
+        [InlineData(typeof(CodeRefactoringProvider), typeof(PredefinedCodeRefactoringProviderNames))]
+        [InlineData(typeof(IConfigurationFixProvider), typeof(PredefinedConfigurationFixProviderNames))]
+        public void TestNameMetadataIsInPredefinedNames(Type providerType, Type predefinedNamesType)
+        {
+            var predefinedNames = GetPredefinedNamesFromType(predefinedNamesType);
+
+            var configuration = EditorTestCompositions.EditorFeatures.GetCompositionConfiguration();
+            var exportedProviders = FindComposedPartsWithExport(configuration, providerType.FullName).ToArray();
+
+            var failureMessage = new StringBuilder();
+            failureMessage.AppendLine($"The following providers were exported with a Name not present in Predefined{providerType.Name}Names:");
+            var passLength = failureMessage.Length;
+
+            foreach (var (providerPart, providerExport) in exportedProviders)
+            {
+                if (providerExport.Metadata.TryGetValue("Name", out var name)
+                    && name is string { Length: > 0 }
+                    && !predefinedNames.Contains(name))
+                {
+                    failureMessage.AppendLine(providerPart.Definition.Type.FullName);
+                }
+            }
+
+            Assert.True(failureMessage.Length == passLength, failureMessage.ToString());
+        }
+
+        [Theory]
+        [InlineData(typeof(CodeFixProvider), typeof(PredefinedCodeFixProviderNames))]
+        [InlineData(typeof(CodeRefactoringProvider), typeof(PredefinedCodeRefactoringProviderNames))]
+        [InlineData(typeof(IConfigurationFixProvider), typeof(PredefinedConfigurationFixProviderNames))]
+        public void TestAllPredefinedNamesUsedAsNameMetadata(Type providerType, Type predefinedNamesType)
+        {
+            var predefinedNames = GetPredefinedNamesFromType(predefinedNamesType);
+
+            var configuration = EditorTestCompositions.EditorFeatures.GetCompositionConfiguration();
+            var exportedProviders = FindComposedPartsWithExport(configuration, providerType.FullName);
+            var providerNames = exportedProviders.Select(exportedProvider =>
```

Comment:
nit: This could be a separate function since it's used in multiple tests

Reply:
I'll extract a TryGetExportName function